playbooks/roles/compatibility-layer/tasks/add_overlay.yml

```diff
@@ -0,0 +1,51 @@
+# Add a custom overlay to the Gentoo Prefix installation.
+---
+- name: Install equery command (dependency for the portage module)
+  command:
+    cmd: emerge gentoolkit
+    creates: "{{ gentoo_prefix_path }}/usr/bin/equery"
+
+- name: Create repos directory
+  file:
+    path: "{{ gentoo_prefix_path }}/etc/portage/repos.conf"
+    state: directory
+    mode: '0755'
+
+- name: Install eselect-repository
+  portage:
+    package: eselect-repository
+    state: present
+
+# We need git in order to add Gentoo overlays hosted on git repositories.
+- name: Install git
+  portage:
+    package: dev-vcs/git
+    state: present
+
+- name: Check which repositories have been installed
+  command: eselect repository list -i
+  register: repositories_installed
+  changed_when: False
+
+- name: Add custom overlay configuration
+  command:
+    cmd: "eselect repository add {{ custom_overlay_name }} {{ custom_overlay_source }} {{ custom_overlay_url }}"
+  when: custom_overlay_name not in repositories_installed.stdout
+
+- name: Sync the repositories
+  portage:
+    sync: yes
```

Comment:
One last comment... 😬 
I think this `yes` should be written as `'yes'`, because the `sync` accepts a string, not a boolean (https://docs.ansible.com/ansible/latest/modules/portage_module.html).

Reply:
Nice catch :-)

I would have expected this to work, since the module defines (and hence interprets?) these parameters as strings. Even one of the examples in their documentation uses `sync: yes` without quotes. But I just changed the value to `no`, and then Ansible warns me about converting `False` to `'False'`. So I guess Ansible does first interpret the values (as booleans), and then passes them to the portage modules. 

TLDR: I'll add quotes. Thanks!